services/streammanager/kafka/kafkamanager.go

```diff
@@ -354,78 +379,73 @@ func Produce(jsonData json.RawMessage, producer interface{}, destConfig interfac
 		return makeErrorResponse(err) //returning 500 for retrying, in case of bad configuration
 	}
 
-	topic := conf.Topic
+	if conf.Topic == "" {
+		return makeErrorResponse(fmt.Errorf("invalid destination configuration: no topic"))
+	}
 
+	ctx := context.TODO()
```

Comment:
I fear we are using too many contexts without it being necessary. Why do we need a context in `kafkaclient.Producer.Publish()` since we set the publish timeout in its configuration?

Reply:
Context is not only used for timeout, the main concept behind context is being able to cancel IO/blocking operations in a hierarchical way.

As a rule of thumb, I would say every method/function should have a context, except if it non-blocking IO and we don't expect it to be in the future.

Reply:
Yes I agree with @lvrach:
> As a rule of thumb, I would say every method/function should have a context, except if it non-blocking IO and we don't expect it to be in the future.